app/models/dataload_job.rb

```diff
@@ -0,0 +1,26 @@
+class DataloadJob < ApplicationRecord
+  belongs_to :vendor
+  enum :status, [
+    :scheduled, # the initial value when a job is created
+    :queued, # the status set when the API has reported the job
+    :in_progress, # when airflow has started the job it will update this value via the API
+    :failed, # if the airflow job fails for any reason, it should update this value via the API
+    :completed # when the job completes successfully in airflow, this value will be updated via the API
+  ]
+  enum :recurrence, [
+    :no_recurrence,
+    :daily_recurrence,
+    :weekly_recurrence,
+    :monthly_recurrence,
+    :yearly_recurrence
+  ]
```

Comment:
Consider declaring these as hashes rather than arrays so re-ordering them won't muck up the values.

Reply:
> Consider declaring these as hashes rather than arrays so re-ordering them won't muck up the values.

can you do an enum in a model as a hash?  Also - since it's an enum, why would the order matter?

Reply:
Yes, and the Rails docs recommend this practice IIRC. The order matters because the values in the enum are assigned sequential integers in the DB. This is another reason why the hash structure is worth considering: it surfaces the DB-persisted values in the model.

Reply:
Done.